lib/post.rb

```diff
@@ -0,0 +1,35 @@
+require 'pg'
+
+class Post
+  attr_reader :id, :date, :author, :message
+
+  def initialize(id:, date:, author:, message:)
+    @id = id
+    @date = date
+    @author = author
+    @message = message
+  end
+
+  def self.all
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+    result = connection.exec("SELECT * FROM peeps ORDER BY date DESC")
+    result.map do |peep|
+      Post.new(id: peep['id'], date: peep['date'], author: peep['author'], message: peep['message'])
+    end
+  end
+
+  def self.create(date:, author:, message:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec_params("INSERT INTO peeps (date, author, message) VALUES('#{date}', $1, $2) RETURNING id, date, author, message;", [author, message])
```

Comment:
Nice that you are [using exec_params to prevent sql injections](https://github.com/makersacademy/course/blob/main/apprenticeships_bookmark_manager/walkthroughs/12.md) here, but I'm wondering why you are not also doing that for the date, especially since the date is sent by the front-end as a param.

It would be different if the date was automatically added in ruby to peeps when they are created, and not coming from the front-end.

I know you form enforces a format for the input, but a malicious user could easily change the front-end html and send data in a different way, so your server should always protect itself from the front-end.